File: Backend/ServicioJobs.Aplicacion/Feature/JobProgramados/Command/AgregarJobProgramado/AgregarJobProgramadoComand.cs
using MediatR;
using ServicioJobs.Modelos.Enums;
using ServicioJobs.Modelos.Utilitarios;

namespace ServicioJobs.Aplicacion.Feature.Programados.Command.JobProgramados
{
    public class AgregarJobProgramadoComand : IRequest<RespuestaServicio<Unit>>
    {
        public string Nombre { get; set; }
        public string Descripcion { get; set; }
        public string Url { get; set; }
        public List<JobParametroItem> JobParametro { get; set; } = new();
        public string Crontab { get; set; }
        public string CorreoNotificar { get; set; }
        public int? ReintentosPermitidos { get; set; }
        public int? PeriodoReintento { get; set; }
        public int? Timeout { get; set; }
        public MetodoHttp MetodoHttp { get; set; }
    }

    public class JobParametroItem
    {
        public string Propiedad { get; set; } = string.Empty;
        public string Valor { get; set; } = string.Empty;
        public TipoParametro Tipo { get; set; }
    }
}

File: FrontEnd/src/Core/Dominio/Model/JobProgramado/AgregarJobProgramadoComand.ts
import { MetodoHttp } from "../enum/MethodoHTTP";
import { JobParametro } from "./JobParametro";

export interface AgregarJobProgramadoComand {
    idMetodo: string;
    nombre: string;
    descripcion: string;
    url: string;
    jobParametro: JobParametro[];
    crontab: string;
    correoNotificar: string;
    reintentosPermitidos?: number;
    periodoReintento?: number;
    timeout?: number;
    metodoHttp: MetodoHttp;
  }
  